### DISCUSSION TITLE
Skip error

### DISCUSSION THREAD

**Discussion by 2V3EvG4LMJFdRe** [beetbox/beets#3960] [Q&A]
Sorry, this feels obvious but it's been hard to search. Is there a flag I can append so that beets doesn't stop on an error? For example, `beet fetchart` will stop when a folder is not found, when I'd prefer it to keep going.

**Comment by jackwilsdon** [comment]
Can you give an example of the folder not found error you're seeing?

**Reply by 2V3EvG4LMJFdRe** [reply]
Here is how they typically look:

```
fetchart: Ayumi Hamasaki - A Best: found album art
fetchart: Ayumi Hamasaki - A Song for ××: found album art
fetchart: Ayumi Hamasaki - ayu-mi-x: found album art
fetchart: Ayumi Hamasaki - ayu-mi-x II version Acoustic Orchestra: found album art
fetchart: Ayumi Hamasaki - ayu-mi-x II version Non-Stop Mega Mix: found album art
fetchart: Ayumi Hamasaki - ayu-mi-x II version US+EU: found album art
fetchart: Ayumi Hamasaki - Duty: found album art
fetchart: Ayumi Hamasaki - I am...: found album art
fetchart: Ayumi Hamasaki - LOVEppears: found album art
fetchart: Ayumi Hamasaki - MY STORY: found album art
fetchart: Ayumi Hamasaki - RAINBOW: found album art
fetchart: Ayumi Hamasaki - SUPER EUROBEAT presents ayu-ro mix: found album art
Error: No such file or directory while moving /var/folders/4y/96gd8p814dq0mlqc24_6_bq00000gn/T/tmpbo2qq4h3.jpg to /Volumes/Expansion-1/Media/Audio/Sound/Ayumi Ishida/Our Connection/cover.jpg
redacted-iMac:~ redacted$ 
```

These errors are always right in the sense that they report an anomaly, such as a folder listed on the .db not being there any more. Still, it's human to not have the .db synced exactly, so this is a problem. For me, in this case, I have a really big library and it took minutes for `fetchart` to get to this folder. Now I'm gonna have to fix it somehow, and then run it again. But then, I'll find another error. So say I find 10 of these from A to Z, that's gonna be a lot of backtracking.

I thought using `-q` would work, but I see now it's only for `import`.

**Comment by wisp3rwind** [comment]
> Sorry, this feels obvious but it's been hard to search. Is there a flag I can append so that beets doesn't stop on an error? 

No, there isn't.

> These errors are always right in the sense that they report an anomaly, such as a folder listed on the .db not being there any more. Still, it's human to not have the .db synced exactly, so this is a problem.

This doesn't necessarily preclude implementing a `--keep-going` flag, but beets generally _does_ make the assumption that it has full and exclusive control over the library. You will invariably experience such errors if you manually remove music from you library without telling beets about it.

> For me, in this case, I have a really big library and it took minutes for fetchart to get to this folder. Now I'm gonna have to fix it somehow, and then run it again. But then, I'll find another error.

Unless you force looking for album art on albums which already have artwork, fetchart should skip albums quickly (maybe unless the filesystem is really slow, such as for a network drive). If that takes a long time, that should probably be considered to be a bug.

**Reply by 2V3EvG4LMJFdRe** [reply]
I think it would be a good thing to add that `--keep-going` flag. Maybe I could form the habit to inform beets upon changes, I'll try. What command should I use to update the db data on a given artist? I think that'd be the easier for me because what I tend to do works within that hierarchy (removing a folder inside an artist, renaming some song titles, etc). Some command like `beet -update "the beatles"`.

Yes, I've indeed removed all album art completely, and have started from 0 with fetchart. I've solved it for now by removing the .db and importing it again as-is.

**Reply by wisp3rwind** [reply]
That command actually exists: https://beets.readthedocs.io/en/stable/reference/cli.html#update
